### ISSUE TITLE
 PmxExport.dll bug

### ISSUE THREAD

**Issue by X3gold** [CazzoPMX/KKPMX#3]
Hello, I noticed a bug in the PmxExport.dll that not all figures are exported. There are mostly only a json file and in the extra folder with a ColorMask.png in it.
![14-04-2022_01-03-36](https://user-images.githubusercontent.com/38250861/163286436-1ca3097f-a8bd-4cf7-9de4-1bbcc43fbabf.jpg)
![14-04-2022_01-05-56](https://user-images.githubusercontent.com/38250861/163286449-bc3e2bec-d077-4aa7-a6bb-a3b4da1a20cd.jpg)
I have then the PMXRedux2.dll of (https://github.com/Snittern/KoikatsuPmxExporterReverseEngineered) and Grey's mesh exporter 1.5.5 of (https://github.com/FlailingFog/KK-Blender-Porter-Pack/blob/9fcef4127ba56b4e8e8718fb546945fc00eaaad9/GME/Grey%27s%20mesh%20exporter%201.5.5.7z) and your old 1.5.0 GeneratorPlugin.dll with it went then also these figures besides were the figures with it no more in the T pose and a further bug I still in the control panel noticed.
![14-04-2022_01-24-43](https://user-images.githubusercontent.com/38250861/163286518-28e61c95-3678-43e1-b505-9b5bc43880e6.jpg)
With the PMXRedux2.dll plugin
![14-04-2022_01-25-56](https://user-images.githubusercontent.com/38250861/163286526-eccf4755-0a8f-4140-affd-16ec8856eaf9.jpg)
With your PmxExport.dll plugin

Under this link are three examples packed in a zip file which do not work with your PmxExport.dll but with PMXRedux2.dll.
https://drive.google.com/file/d/14I8rLnlJyQA1R5RE0jFACDzZ_Oh4s9lo/view

**Comment by CazzoPMX** [comment]
Uh..... did you per chance use KK Sunshine? I haven't found the time yet to get the texture export working there so I disabled it in that version. I'll look into it when I find time.

**Comment by CazzoPMX** [comment]
Well, I cannot seem to replicate this issue (I also use KK Party, mind you). Maybe you could send me your KK Log file through DM or something? It should contain Exceptions if something went wrong.

Although I didn't base off my version of PMXRedux (I decompiled it myself after only finding a broken version), it certainly causes the buttons to overlay and only the PMXRedux triggers (which does not export any [extras] textures).

The Generator Plugin adds a Button in the CharacterTab which only creates a JSON file and has since then been merged into my PMXExport version, so its kinda obsolete.

I also don't do anything with the control panel (or any other UI Element for that matter), so I find it weird that it affects it.
Which plugin makes that btw?

**Comment by CazzoPMX** [comment]
Oh my, that Animator Commit from PMXRedux apparently caused these T-Pose issues to disappear at least \o/

**Comment by X3gold** [comment]
For example, these cards do not go with your plugin oh yes I also use only KK Party.
![93273232_p7](https://user-images.githubusercontent.com/38250861/165998213-945bde47-a78b-4386-88f6-ee03af9e8a8a.png)
![89469987_p2](https://user-images.githubusercontent.com/38250861/165998246-39ae65b6-efcb-45f9-81c1-17d89c9a7eb2.png)
![89469987_p3](https://user-images.githubusercontent.com/38250861/165998274-5e6dec2b-d062-4c1d-a7cb-bfabe852cccf.png)
![89469987_p4](https://user-images.githubusercontent.com/38250861/165998288-ed4a7c7f-c67d-48e5-bc1c-d566739fdc1f.png)

**Comment by CazzoPMX** [comment]
They all export properly for me..... You sure it is because of that specific plugin? Does your "output_log.log" show any errors or exceptions?

**Comment by X3gold** [comment]
I took the Koikatsu BetterRepack RX13 (https://dl.betterrepack.com/public/#list-koikatsu) again, with it your plugin had with me unfortunately still this export bug.
[output_log.txt](https://github.com/CazzoPMX/KKPMX/files/8597406/output_log.txt)

**Comment by X3gold** [comment]
The plugin is called ClothingStateMenuX.Koikatu.dll (https://github.com/IllusionMods/KeelPlugins)
but you also need KKAPI.dll and KKAPI.xml (https://github.com/IllusionMods/IllusionModdingAPI) and ExtensibleSaveFormat.dll and ExtensibleSaveFormat.xml (https://github.com/IllusionMods/BepisPlugins) as I found out.

**Comment by X3gold** [comment]
I didn't notice that you had updated to version 1.6.2, with this version the bugs described above are gone, I have no more problems with the export and the menu is also displayed correctly.

**Comment by CazzoPMX** [comment]
Happy to hear that. And I also saw (probably) the reason for your weird behaviour: the output log you supplied contains a NullReferenceException (aka something is empty when it shouldn't) with regards to AnimatorPoses. And the PmxRedux had a commit 5 months ago which fixed an issue regarding Animators. I added that same fix in my version too. This might have caused some textures to not load correctly, which led to an issue during the export (which is now catched gracefully as well).

I do have all these plugins in question, but I didn't knew that the ClothingState got an update.
If you have no other problems regarding this, I will close this issue then.
